generated/nifake/nifake/unit_tests/test_converters.py

```diff
@@ -237,6 +237,115 @@ def test_repeated_capabilies_without_prefix():
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev2/2'
+
+
+def test_resource_name_unicode_string():
+    test_result_list = _converters.convert_resource_name(u'Dev1')
+    assert test_result_list == 'Dev1'
+    test_result_list = _converters.convert_resource_name(u'Dev1,Dev2')
+    assert test_result_list == 'Dev1,Dev2'
+    test_result_list = _converters.convert_resource_name(u'Dev1/0')
+    assert test_result_list == 'Dev1/0'
+    test_result_list = _converters.convert_resource_name(u'Dev1/0:1, Dev2/0-1')
+    assert test_result_list == 'Dev1/0,Dev1/1,Dev2/0,Dev2/1'
+    test_result_list = _converters.convert_resource_name(u'Dev1, Dev2/0-2')
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev2/2'
+
+
+def test_resource_name_raw_string():
+    test_result_list = _converters.convert_resource_name(r'Dev1')
+    assert test_result_list == 'Dev1'
+    test_result_list = _converters.convert_resource_name(r'Dev1,Dev2')
+    assert test_result_list == 'Dev1,Dev2'
+    test_result_list = _converters.convert_resource_name(r'Dev1/0')
+    assert test_result_list == 'Dev1/0'
+    test_result_list = _converters.convert_resource_name(r'Dev1/0:1, Dev2/0-1')
+    assert test_result_list == 'Dev1/0,Dev1/1,Dev2/0,Dev2/1'
+    test_result_list = _converters.convert_resource_name(r'Dev1, Dev2/0-2')
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev2/2'
+
+
+def test_resource_name_list():
+    test_result_list = _converters.convert_resource_name(['Dev1'])
+    assert test_result_list == 'Dev1'
+    test_result_list = _converters.convert_resource_name(['Dev1', 'Dev2'])
+    assert test_result_list == 'Dev1,Dev2'
+    test_result_list = _converters.convert_resource_name(['Dev1/0'])
+    assert test_result_list == 'Dev1/0'
+    test_result_list = _converters.convert_resource_name(['Dev1/0:1', 'Dev2/0-1'])
+    assert test_result_list == 'Dev1/0,Dev1/1,Dev2/0,Dev2/1'
+    test_result_list = _converters.convert_resource_name(['Dev1', 'Dev2/0-2'])
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev2/2'
+
+
+def test_resource_name_tuple():
+    test_result_list = _converters.convert_resource_name(('Dev1'))
+    assert test_result_list == 'Dev1'
+    test_result_list = _converters.convert_resource_name(('Dev1', 'Dev2'))
+    assert test_result_list == 'Dev1,Dev2'
+    test_result_list = _converters.convert_resource_name(('Dev1/0'))
+    assert test_result_list == 'Dev1/0'
+    test_result_list = _converters.convert_resource_name(('Dev1/0:1', 'Dev2/0-1'))
+    assert test_result_list == 'Dev1/0,Dev1/1,Dev2/0,Dev2/1'
+    test_result_list = _converters.convert_resource_name(('Dev1', 'Dev2/0-2'))
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev2/2'
+
+
+def test_resource_name_mixed():
+    test_result_list = _converters.convert_resource_name(['Dev1', ('Dev2/0', 'Dev2/1'), ['Dev3/0:1', 'Dev4/1:2'], 'Dev5/1'])
+    assert test_result_list == 'Dev1,Dev2/0,Dev2/1,Dev3/0,Dev3/1,Dev4/1,Dev4/2,Dev5/1'
+
+
+def test_invalid_resource_name():
```

Comment:
Any reason not to use pytest.raises?
https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions